test/apify/crawlers/basic_crawler.test.ts

```diff
@@ -641,6 +664,198 @@ describe('BasicCrawler', () => {
         expect(crawler.handleRequestTimeoutMillis).toBe(maxSignedInteger);
     });
 
+    test('should not log stack trace for timeout errors by default', async () => {
+        const sources = [{ url: `http://${HOSTNAME}:${port}` }];
+        const requestList = await RequestList.open(null, sources);
+
+        const crawler = new BasicCrawler({
+            requestList,
+            handleRequestTimeoutSecs: 0.005,
+            maxRequestRetries: 3,
+            handleRequestFunction: () => sleep(10),
+        });
+
+        const warnMock = jest.fn();
+        // @ts-expect-error Overriding protected method
+        crawler.log.warning = warnMock;
+
+        const errMock = jest.fn();
+        // @ts-expect-error Overriding protected method
+        crawler.log.error = errMock;
```

Comment:
Can't you use jest.spyOn instead? `const warningSpy = jest.spyOn(crawler.log, 'warning')` etc

Reply:
it's pretty much the same, but yeah - I guess it's more readable/clean. Updated.